customskill/response/response.go

```diff
@@ -0,0 +1,114 @@
+package response
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+var jsonMarshal = json.Marshal // Used to enable unit testing
+
+func New() *Envelope {
+	e := &Envelope{
+		Version: "1.0",
+		Response: Response{
+			ShouldEndSession: true,
+		},
+		SessionAttributes: make(map[string]interface{}),
+	}
+
+	return e
+}
+
+func (e *Envelope) SetOutputSpeech(text string) *Envelope {
+	e.Response.OutputSpeech = &OutputSpeech{
+		Type: "PlainText",
+		Text: text,
+	}
+
+	return e
+}
+
+func (e *Envelope) SetOutputSpeechSSML(text string) *Envelope {
+	e.Response.OutputSpeech = &OutputSpeech{
+		Type: "SSML",
+		SSML: text,
+	}
+
+	return e
+}
+
+func (e *Envelope) SetReprompt(text string) *Envelope {
+	e.Response.Reprompt = &Reprompt{
+		OutputSpeech: &OutputSpeech{
+			Type: "PlainText",
+			Text: text,
+		},
+	}
+
+	return e
+}
+
+func (e *Envelope) SetRepromptSSML(text string) *Envelope {
+	e.Response.Reprompt = &Reprompt{
+		OutputSpeech: &OutputSpeech{
+			Type: "SSML",
+			SSML: text,
+		},
+	}
+
+	return e
+}
+
+func (e *Envelope) SetCard(title string, content string) *Envelope {
+	return e.SetSimpleCard(title, content)
+}
+
+func (e *Envelope) SetSimpleCard(title string, content string) *Envelope {
+	e.Response.Card = &Card{
+		Type:    "Simple",
+		Title:   title,
+		Content: content,
+	}
+
+	return e
+}
+
+func (e *Envelope) SetStandardCard(title string, content string, smallImg string, largeImg string) *Envelope {
```

Comment:
for functions with the same parameter type you can just specify the type at the end of the list something like:

SetStandardCard(title, content, smallImg, largImg string) *Envelope

Reply:
good catch. I will go back and consolidate instances of this